soroban-sdk-macros/src/derive_fn.rs

```diff
@@ -150,6 +151,9 @@ pub fn derive_pub_fn(
                 env: #crate_path::Env,
                 args: &[#crate_path::Val],
             ) -> #crate_path::Val {
+                if args.len() != #arg_count {
+                    panic!("invalid number of input arguments: {} expected, got {}", #arg_count, args.len());
```

Comment:
Love this 👏🏻

Reply:
Actually, this a pretty serious bug in the test utilities because if you invoke a natively registered contract with too many args the additional args just get ignored. I'd like us to fix this in the current SDK v21. I've opened a PR with your fix extracted in isolation and added tests:
- https://github.com/stellar/rs-soroban-sdk/pull/1336

I'd like to release that fix in a v21 patch release once merged.